Controllers/ShoppingCartController.cs

```diff
@@ -382,10 +379,21 @@ public class ShoppingCartController : Controller {
         }
 
         private void UpdateShoppingCart(IEnumerable<UpdateShoppingCartItemViewModel> items) {
+            var oldItems = new List<ShoppingCartItem>();
+            oldItems.AddRange(_shoppingCart.Items); //create a copy for analysis
             _shoppingCart.Clear();
 
-            if (items == null)
+            if (items == null) {
+                //removed all items
+                foreach (var han in _cartLifeCycleEventHandlers) {

```

Comment:
I am coding on a Lettera 32, and writing han saves me ink.
(I changed this to `handler` through all the code)